application/src/main/java/org/togetherjava/tjbot/commands/system/JDAScheduler.java

```diff
@@ -0,0 +1,34 @@
+package org.togetherjava.tjbot.commands.system;
+
+import net.dv8tion.jda.api.JDA;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.concurrent.*;
+
+public class JDAScheduler {
+    private static JDAScheduler instance;
+
+    private final JDA jda;
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    private JDAScheduler(JDA jda) {
+        this.jda = jda;
+    }
+
+    public static void create(@NotNull JDA jda) {
+        Objects.requireNonNull(jda, "Cannot create a JDAScheduler without an instance of JDA");
+        if (instance != null) {
+            throw new IllegalStateException("Can only create one instance of JDAScheduler");
+        }
+
+        instance = new JDAScheduler(jda);
+    }
+
+    public static ScheduledFuture scheduleTask(Callable task, long delay, TimeUnit timeUnit) {
+        Objects.requireNonNull(instance,
+                "Cannot schedule a task before the scheduler has been created");
+
+        return instance.scheduler.schedule(task, delay, timeUnit);
+    }
+}
```

Comment:
Honestly, I would suggest to move this into your package as "private solution" until the full solution is out.

Reply:
This class has some questionable design and I dont want it to be used by others before #235 is done. So either you have to fix all the design issues and make this super nice (i.e. implement #235 and ship it separately before this) or you just make this your own little "private" helper until #235 is shipped.

Reply:
yeah, this should be in these commits, i will delete it

i was 'feeling' my way towards a solution, this is still extremely early
i stopped because #235 was added as an issue
(and because u said first ship the replacement then improve on it)

the scheduler is not being used anywhere currently and its not in a state thats anywhere near functional yet

it definitely should not be merged into develop but should it get stashed in another branch for now?
or just deleted

Reply:
you can definitely move it into a separate branch, sure.

Reply:
fixed